client/src/main/java/io/pravega/client/netty/impl/FlowHandler.java

```diff
@@ -308,7 +314,7 @@ public void close() {
         public void run() {
             try {
                 if (!recentMessage.getAndSet(false)) {
-                    Futures.getAndHandleExceptions(getChannel().writeAndFlush(new WireCommands.KeepAlive()), ConnectionFailedException::new);
+                    getChannel().write(new WireCommands.KeepAlive()).addListener(ChannelFutureListener.CLOSE_ON_FAILURE);
```

Comment:
It actually makes no difference because I have modified the command encoder to flush in response to a keep alive. 
I removed it here because it would mean doing it twice. I could reverse the order and do it here and not in command encoder, if you think that would be more clear.

Reply:
The latest code fixes it. Resolving this comment.